devel/lazarus/Portfile

```diff
@@ -0,0 +1,127 @@
+
+# adjust the default paths to MacPorts paths
+    set patchtarget ${worksrcpath}/tools/install/macosx/environmentoptions.xml
+    reinplace "s|/usr/local/bin/_PPCARCH_|${prefix}/bin/ppcx64|g" ${patchtarget}
+    reinplace "s|/usr/local/share|${prefix}/share|g"              ${patchtarget}
+    reinplace "s|/Developer/lazarus|${prefix}/share/lazarus|g"    ${patchtarget}
+
+    reinplace "s|<DebuggerFilename Value=\"/usr/bin/lldb\"/>|<DebuggerFilename Value=\"/usr/bin/lldb\"> <History Count=\"3\"> <Item1 Value=\"/usr/bin/lldb\"/> <Item2 Value=\"/usr/bin/gdb\"/> <Item3 Value=\"${prefix}/bin/gdb\"/> </History> </DebuggerFilename>|g" ${patchtarget}
+# fix path to X11
+    reinplace "s|-Fl/usr/X11R6/lib -Fl/sw/lib|\"-Fl${prefix}/lib -Fl${prefix}/lib/pango-ft219/lib -Fl/opt/X11/lib\"|g" ${worksrcpath}/ide/Makefile
+
+# fix compilation of bigidecomponents with LCL_PLATFORM=nogui (LCLnogui)
+# with osprinters.pas the line number needs to be given
+# Note: I did not manage a solution with reinplace
+    system -W ${worksrcpath}/components/printers \
+        "sed -i.tmp '50 s|LCLGtk2|LCLnogui\}\{\$I cupsprinters_h.inc\}\{\$ENDIF\}\{\$IFDEF LCLGtk2|g' osprinters.pas && \
+         sed -i.tmp '87 s|LCLGtk2|LCLnogui\}\{\$I cupsprinters.inc\}\{\$ENDIF\}\{\$IFDEF LCLGtk2|g'   osprinters.pas && \
+         sed -i.tmp 's|LCLGtk2|LCLnogui\}uses udlgSelectPrinter, udlgPropertiesPrinter, udlgPageSetup;\{\$I cupsprndialogs.inc\}\{\$ENDIF\}\{\$IFDEF LCLGtk2|g' ${worksrcpath}/components/printers/printersdlgs.pp "
+
+# some more fixes for nogui taking cocoa as a template
+# this may break, as soon as cocoa gets working versions
+    copy       ${worksrcpath}/components/lclextensions/include/cocoa ${worksrcpath}/components/lclextensions/include/nogui
+    reinplace "s|CocoaInt, CocoaGDIObjects, ||g"                     ${worksrcpath}/components/lclextensions/include/nogui/uses.inc
+    reinplace "s|Cocoa Interface|NoGui Interface|g"                  ${worksrcpath}/components/lclextensions/include/nogui/delphicompat.inc
+
+# Note: I did not manage a solution with reinplace
+    system -W ${worksrcpath}/components/lclextensions/include/nogui \
+        "sed -i.tmp 's|\$define HAS_GETCURRENTOBJECT|.\$define HAS_GETCURRENTOBJECT|g' delphicompat.inc && \
+         sed -i.tmp '21,27d' delphicompat.inc && \
+         sed -i.tmp '46,68d' delphicompat.inc "
+
+    copy ${worksrcpath}/components/virtualtreeview/units/cocoa         ${worksrcpath}/components/virtualtreeview/units/nogui
+    copy ${worksrcpath}/components/virtualtreeview/include/intf/cocoa/ ${worksrcpath}/components/virtualtreeview/include/intf/nogui
+
+}
+
+build.target       bigide
+build.post_args    OPT=-gl LCL_PLATFORM=cocoa
+use_parallel_build no
+
+destroot {
+    xinstall -m 755 -d ${destroot}${prefix}/etc/lazarus
+    xinstall -m 644 ${worksrcpath}/tools/install/macosx/environmentoptions.xml ${destroot}${prefix}/etc/lazarus
+
+    system -W ${worksrcpath} "make install INSTALL_PREFIX=${destroot}${prefix}"
```

Comment:
This is the default destroot command, so don't override the destroot phase. Instead set this:

`destroot.post_args  INSTALL_PREFIX =${destroot}${prefix}
`

You can put the rest of the commands in either `pre-destroot` or `post-destroot` blocks.

Reply:
OK.